main/io.home-assistant/fan.js

```diff
@@ -51,8 +51,8 @@ module.exports = class HomeAssistantFan extends HomeAssistantDevice {
 
     async do_set_oscillation({ oscillation }) {
         if (oscillation === 'on')
-            await this._callService('fan', 'oscillating', true);
+            await this._callService('fan', { 'oscillating': true });
         else
-            await this._callService('fan', 'oscillating', false);
+            await this._callService('fan', { 'oscillating': false });
```

Comment:
This is _still_ not correct: the second argument of _callService needs to be a string. See the implementation at https://github.com/stanford-oval/thingpedia-common-devices/blob/master/main/io.home-assistant/base.js#L37

It maps to HomeAssistant.callService which takes the domain (string), service (string) and data (object).